cmake/kokkos_arch.cmake

```diff
@@ -956,6 +959,11 @@ ENDIF()
 
 FOREACH(ARCH IN LISTS SUPPORTED_AMD_ARCHS)
   IF (KOKKOS_ARCH_${ARCH})
+    STRING(REGEX MATCH "942" IS_942 ${ARCH})
+    IF(IS_942)
+      SET(KOKKOS_ARCH_AMD_GFX942 ON)
+      BREAK()
+    ENDIF()
```

Comment:
I don't think this is necessary. The reason it's there for the other architectures is because I don't know if the user used `AMD_GFX` or `VEGA` but whatever his choice, we want to export both macros. That's why I check for the common part of the flag and then set both variables. I am 99% sure you can remove this but you should double check.

Reply:
Ah, yeah -- that tracks.

Reply:
Correct, removed in [18c88f7](https://github.com/kokkos/kokkos/pull/6358/commits/18c88f700c4348e21fe00c34445940f91d6af4c8)